http.go

```diff
@@ -128,24 +141,44 @@ type responseValue struct {
 }
 
 type responseWriter struct {
-	http.ResponseWriter
+	w           http.ResponseWriter
 	wroteHeader bool
 	code        int
 	bytes       int
 	tee         io.Writer
 }
 
+// Header returns the header map except for a list of filtered-out headers
+// defined by the stripOutHeaders array.
+func (b *responseWriter) Header() http.Header {
```

Comment:
My first approach was to remove those header before building `responseValue`, that worked for all the requests except for the first one. My second approach was to build and re-use a filtered list of headers, and that didn't work as expected because at some point in the request that filtered map was modified by the caller and the bad headers were added. We cannot prevent the modification of this map, the best thing I could come out with is a runtime filter that prevents specific keys from being returned, even if they're present. We can optimize this a bit and check for the existence of the bad keys before allocating a new `http.Header`.

Reply:
I found another way that doesn't require a runtime filter not a new allocation.